Source/Additions/UIControl+LIFEAdditions.m

```diff
@@ -0,0 +1,99 @@
+//
+//  UIControl+LIFEAdditions.m
+//  Buglife
+//
+//  Created by Daniel DeCovnick on 1/17/18.
+//  Copyright © 2018 Buglife, Inc. All rights reserved.
+//
+
+#import "UIControl+LIFEAdditions.h"
+#import "LIFEMacros.h"
+#import <objc/runtime.h>
+#import "LIFEAwesomeLogger+Protected.h"
+#import "LIFEContextAwareLogFormatter.h"
+#import "Buglife.h"
+
+
+@implementation UIControl (LIFEAdditions)
+LIFE_CATEGORY_METHOD_IMPL(UIControl)
+
++ (void)swizzleSendAction
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (Buglife.sharedBuglife.captureUserEventsEnabled) {
+            
+            static dispatch_once_t onceToken;
+            dispatch_once(&onceToken, ^{
+                
+                Method m = class_getInstanceMethod(self, @selector(sendAction:to:forEvent:));
+                if (method_getImplementation(m) == (IMP)life_sendActionImp)
+                {
+                    return;
+                }
+                __originalSendActionImp = method_setImplementation(m, (IMP)life_sendActionImp);
+            });
+        }
+    });
+}
++ (void)load
+{
+    [self swizzleSendAction];
+}
+static IMP __originalSendActionImp;
+static void life_sendActionImp(id self, SEL _cmd, SEL action, id target, UIEvent *event)
+{
+    if (Buglife.sharedBuglife.captureUserEventsEnabled)
+    {
+        if ([self isKindOfClass:[UIButton class]])
+        {
+            if ([[self titleLabel].text isEqualToString:@""])
+            {
+                [LIFEAwesomeLogger.sharedLogger _logDebugMessage:[NSString stringWithFormat:@"Button tapped: %@", self] context:LIFELogContextUserEvent];
+            }
+            else
+            {
+                [LIFEAwesomeLogger.sharedLogger _logDebugMessage:[NSString stringWithFormat:@"Button tapped: %@", [self titleLabel].text] context:LIFELogContextUserEvent];
+
+            }
+        }
+        //UISegmentedContrl
+        //UISlider
+        //UISwitch
+        //UIDatePicker
+        //UIPageControl
+        //UITextField
+        /* //Turn these back on as they're tested.
```

Comment:
There's some commented out code here. I don't have a strong opinion on whether this is kosher to push so I'll leave it up to you (I tend to do the same thing when coding solo)